lib/decimal.ex

```diff
@@ -1391,6 +1391,26 @@ defmodule Decimal do
     tmp
   end
 
+  @doc """
+  Returns `true` when the given `decimal` has no significant digits after the decimal point.
+
+  ## Examples
+
+      iex> Decimal.integer?("1.00")
+      true
+
+      iex> Decimal.integer?("1.1000")
+      false
+  """
+  @spec integer?(t) :: boolean
```

Comment:
since we're passing the input through `|> decimal()` and thus accept integers & strings, the type should be `decimal()`

```suggestion
  @spec integer?(decimal()) :: boolean
```